src/Client.php

```diff
@@ -188,23 +194,47 @@ public function getOptions()
     }
 
     /**
-     * Creates a new client instance for the specified connection ID or alias,
-     * only when working with an aggregate connection (cluster, replication).
-     * The new client instances uses the same options of the original one.
+     * Creates a new client from the specified .
      *
-     * @param string $connectionID Identifier of a connection.
+     * The new client instances inherites the same options of the original one.
+     * When no callable object is supplied, this method returns the new client.
+     * When a callable object is supplied, the new client is passed as its sole
+     * argument and its return value is returned by this method to the caller.
      *
-     * @throws \InvalidArgumentException
+     * NOTE: This method works against any kind of underlying connection object
+     * as it uses a duck-typing approach and looks for a suitable method that
+     * matches the selector type to extract the correct connection.
+     *
+     * @param string        $selector Type of selector (`id`, `key`, `slot`, `command`)
+     * @param string        $value    Values of selector.
+     * @param callable|null $callable Optional callable object.
      *
-     * @return Client
+     * @return ClientInterface|mixed
```

Comment:
is it really mixed or rather object at least?

Reply:
I think this should really be just `ClientInterface` and I can't think of a valid reason why `mixed` is actually there.

And to be honest I can't even remember why there's an optional third parameter for this method accepting a callable object, as this case is not even covered by the tests and I can't think of a valid use case for this.